_P201_VBAN.ino

```diff
@@ -0,0 +1,577 @@
+      UDP_IP = IPAddress(PCONFIG_LONG(0));
+      UDP_PORT = PCONFIG(0);
+      AUDIO_OFFSET = PCONFIG(3);
+      
+      
+      
+      if (ismulticast(UDP_IP))
+      {
+        addLog(LOG_LEVEL_INFO, F("IP address is Multicast"));
+        p201UDP.beginMulticast(UDP_IP,UDP_PORT);
+      }
+      
+
+      buffer = (uint8_t*)calloc(P201_BUFFER_SIZE,sizeof(uint8_t));
+      if (buffer == NULL) 
+      {
+        addLog(LOG_LEVEL_ERROR,F("VBAN: Buffer Allocation error"));
+      }
+      else
+      {
+        P201_fillHeader((T_VBAN_HEADER *)buffer);
+        P201_fillHeader((T_VBAN_HEADER *)(buffer+VBAN_PACKET_SIZE));
+      }
+
+      
+      String log = F("VBAN: sending to: ");
+      log+=UDP_IP.toString();
+      log+=F(":");
+      log+=UDP_PORT;
+      log+=F(" stream=");
+      log+=Settings.Name ;
+      addLog(LOG_LEVEL_INFO,log);
+      
+
+      /*
+      Trilevel Amplifier Gain Control.
+       GAIN = VDD, gain set to 40dB. 
+       GAIN = GND, gain set to 50dB. 
+       GAIN = Unconnected, uncompressed gain set to 60dB. 
+      */
+      if (CONFIG_PIN2>0)
+       switch(PCONFIG(1)){
+          case 1: //40db
+            pinMode(CONFIG_PIN2,OUTPUT);
+            digitalWrite(CONFIG_PIN2,HIGH);
+            break;
+          case 2: //50db
+            pinMode(CONFIG_PIN2,OUTPUT);
+            digitalWrite(CONFIG_PIN2,LOW);
+            break;
+          case 3: //60db
+          default:
+          pinMode(CONFIG_PIN2, INPUT); //high Z
+        }
+
+      /*
+      Trilevel Attack and Release Ratio Select. Controls the ratio of attack time to release time for the AGC circuit.
+       A/R = GND: Attack/Release Ratio is 1:500 
+       A/R = VDD: Attack/Release Ratio is 1:2000 
+       A/R = Unconnected: Attack/Release Ratio is 1:4000
+      */
+      if (CONFIG_PIN3>0)
+        switch(PCONFIG(2)){
+          case 1: //AR=1:500
+            pinMode(CONFIG_PIN3,OUTPUT);
+            digitalWrite(CONFIG_PIN3,LOW);
+            break;
+          case 2: //AR=1:2000
+            pinMode(CONFIG_PIN3,OUTPUT);
+            digitalWrite(CONFIG_PIN3,HIGH);
+            break;
+          case 3: //60db
+          default:
+          pinMode(CONFIG_PIN3, INPUT); //high Z
+        }
+      
+      //p201UDP.begin(VBAN_PORT);
+
+
+      //WiFi.setTxPower(WIFI_POWER_MINUS_1dBm);
+
+      
+      ADC_CHANNEL = (adc1_channel_t)digitalPinToAnalogChannel(CONFIG_PIN1);
+      
+      {
+      adc1_config_width(ADC_WIDTH_12Bit); // configure the analogue to digital converter
+      adc1_config_channel_atten(ADC_CHANNEL, ADC_ATTENUATION); //ADC_ATTEN_11db); // connects the ADC 1 with channel 7 (GPIO 35)
+      adc1_get_raw(ADC_CHANNEL); //we need this to initialize the ADC correctly
+
+      
+      adc1_config_channel_atten(ADC1_CHANNEL_1, ADC_ATTENUATION); //ADC_ATTEN_11db); // connects the ADC 1 with channel 7 (GPIO 35)
+      adc1_get_raw(ADC1_CHANNEL_1); //we need this to initialize the ADC correctly

```

Comment:
You do call adc1 functions, but it is possible to select a GPIO for the ADC2 input. (unlikely that you will actually use it as it interferes with WiFi)

Reply:
I will need to filter out selection of ADC2 pins somehow in the future. did not get there yet.
Maybe you know how.

Reply:
Was thinking to just use the channels selection and create a new selection dropdown.

Reply:
also thanks for catching this one as was extra code left over from testing to see if I can measure the noise.

Reply:
> I will need to filter out selection of ADC2 pins somehow in the future. did not get there yet.
> Maybe you know how.

You could add an extra parameter to this pin selector dialog to show:
- All pins capable of analog input  (ESP32-S2 has more, so we also need to extend the list for that one later)
- Pins connected to ADC1
- pins connected to ADC2

So I guess it would be best to use an enum for that.  (default set to "all")

Reply:
I did used a dropdown in the end